src/main/java/org/tugraz/sysds/runtime/data/DenseBlockBool.java

```diff
@@ -155,26 +155,31 @@ public DenseBlock set(int r, int c, double v) {
 	
 	@Override
 	public DenseBlock set(DenseBlock db) {
-		System.arraycopy(db.valuesAt(0), 0, _data, 0, _rlen*_odims[0]);
```

Comment:
I would recommend to run a few experiments (maybe in a separate PR for performance improvements) and see if we could can something if the input DenseBlock is a DenseBlockBool (i.e., _data.clear(); _data.or(that._data))